virttest/utils_test/__init__.py

```diff
@@ -1841,8 +1845,88 @@ def app_running(self):
         """
         session = self.get_session()
         status = session.cmd_status(self.check_cmd, timeout=60)
+        session.close()
         return status == 0
 
+    def _git_download(self, url, destination):
+        """
+        git download
+        :param url: download url
+        :param destination: download destination
+        """
+        from avocado.utils import git
+        destination = os.path.join(destination, self.base_name)
+        git.get_repo(url, destination_dir=destination)
+
+    def _tarball_download(self, url, destination):
+        """
+        tarball download and uncompress
+        :param url: dowload url
+        :param destination: download destination
+        """
+        import tarfile
+        tarball_path = os.path.join(destination, self.base_name)
+        download.get_file(url, tarball_path)
+        tarball = tarfile.open(tarball_path)
+        tarball.extractall(destination)
+        self.base_name = tarball.members[0].name.split("/")[0]
+        tarball.close()
+
+    @error_context.context_aware
+    def download_stress(self):
+        """
+        Download stress tool
+        """
+        url = self.download_url
+        tmp_path = data_dir.get_tmp_dir()
+        error_context.context('Download stress tool from %s' % url,
+                              logging.info)
+        download_method = getattr(self, "_%s_download" % self.download_type)
+        download_method(url, tmp_path)
+        error_context.context('Copy stress tool to guest', logging.info)
+        self.vm.wait_for_login()
+        self.vm.copy_files_to(os.path.join(tmp_path, self.base_name), self.dst_path)
+
+    @error_context.context_aware
+    def install(self):
+        """
+        To download, abstract, build and install the stress tool
+        """
+        self.download_stress()
+        install_path = os.path.join(self.dst_path, self.base_name)
+        session = self.get_session()
+        session.cmd_output_safe('cd %s' % install_path)
+        error_context.context('make and install the %s' %
+                              self.stress_type, logging.info)
+        status, output = session.cmd_status_output(self.make_cmds)
+        session.close()
+        if status != 0:
+            raise exceptions.TestError(
+                "Installation failed with output:\n %s" % output)
+
+    @error_context.context_aware
+    def clean(self):
+        """
+        Uninstall stress application, and clean the source files
+        """
+        session = self.get_session()
+        install_path = os.path.join(self.dst_path, self.base_name)
+        if session.cmd_status('cd %s' % install_path) != 0:
+            error_context.context("No source files found in path %s" % path)
+            session.close()
+            return
+
+        uninstall_cmd = './configure && make uninstall'
```

Comment:
@vivianQizhu uninstall cmd does not require ./configure again and it is better to get from user like `make_cmds`

Reply:
@sathnaga In my test, sometimes `make uninstall` will fail if don't `./configure` first, this might rely on the process between `make uninstall` and `make install`, I wish this function could be more independent, and I supposed that with `./configure` here is harmless, so allow me keep it here. But true it should be configurable, I will move it to init params. Thanks for review.